src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -217,18 +217,17 @@ module.exports = function<T, P, I, TI, C>(
     const prevProps = current ? current.memoizedProps : null;
     let nextChildren = nextProps.children;
     const isDirectTextChild = shouldSetTextContent(nextProps);
+
     if (isDirectTextChild) {
       // We special case a direct text child of a host node. This is a common
       // case. We won't handle it as a reified child. We will instead handle
       // this in the host environment that also have access to this prop. That
       // avoids allocating another HostText fiber and traversing it.
       nextChildren = null;
-    } else if (prevProps && (
-      shouldSetTextContent(prevProps) ||
-      prevProps.children === null ||
```

Comment:
So I think the reason I added these additional checks is to support the case where we switch from a direct text child to an empty child (null, undefined, or boolean). So these checks should actually be comparing `nextProps.children`, not `prevProps.children`.

Could you add a test for these cases? It's my fault those don't already exist :)